src/Symfony/Component/Console/Application.php

```diff
@@ -172,10 +173,13 @@ public function run(InputInterface $input = null, OutputInterface $output = null
 
         try {
             $exitCode = $this->doRun($input, $output);
-        } catch (\Exception $e) {
+        } catch (\Throwable $e) {
             if (!$this->catchExceptions) {
                 throw $e;
             }
+            if (!$e instanceof \Exception && !$this->catchErrors) {
```

Comment:
Instead of the `instanceof`, why not having 2 `catch`?

```php
try {
} catch (\Exception $e) {
    if (!$this->catchExceptions) {
        throw $e;
    }
} catch (\Throwable $e) {
    if (!$this->catchErrors) {
        throw $e;
    }
}
```

Someone may want to catch errors but not exceptions, who knows.

Reply:
But we want the exception handling in the `} catch (\Exception $e) {` part. With your code we loose this feature.

Reply:
Indeed, I didn't pay much attention to that. You could factor the error handling into a function, but that wouldn't be any better. Let's keep it that way.